src/components/types/index.tsx

```diff
@@ -0,0 +1,14 @@
+export type TodoType = {
+  id: number,
+  title: string,
+  completed: boolean,
+  userId: number,
+  user: UserType | null,
+};
+
+export type UserType = {
+  id: number,
+  name: string,
+  username: string,
+  email: string,
+};
```

Comment:
I would add different files for different types. For example, Todo.ts for todo type and User.ts for user type